platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/TlsFactory.java

```diff
@@ -123,4 +132,25 @@ public TlsFactory(
         clientSocket.startHandshake();
         return clientSocket;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void reload(@NonNull final List<PeerInfo> peers) {
+        lock.lock();
```

Comment:
What is the purpose of this lock? This class as a whole is not thread safe if we have multi-threaded access, and currently `reload` is only called from this class's constructor.

Reply:
Before we add locks to all of the methods in this class, it might be helpful to define the access pattern we intend to support with this class. It may be the case that we can guarantee an access pattern that does not require locking.

Reply:
`reload` will eventually be called from the outside of this class, and we do need the operations to be atomic when it does.

>It may be the case that we can guarantee an access pattern that does not require locking
This is true but i don't think it's the case here. This class is doing a lot of long ish operations that may interleave easily (e.g. other threads may easily access the default SSL Context), locking seems the reasonable approach to achieving atomicity.

Reply:
I would not add a lock prematurely, I think we should avoid concurrent access if possible

Reply:
As discussed, this isn't a premature lock. The public access to the static `SSLContext.getDefault()` outside of this class legitimizes a case for locking. @lpetrovic05 proposed exploring the removal of `SSLContext.setDefault()`. This is old code. I'm working on removing it, and subsequently the lock.